File: rsleetcode/src/easy/excel_sheet_column_number.rs
//! 171. Excel Sheet Column Number
//!
//! https://leetcode.com/problems/excel-sheet-column-number/

pub struct Solution {
}

impl Solution {
    pub fn title_to_number(column_title: String) -> i32 {
        column_title
            .into_bytes()
            .into_iter()
            .rev()
            .enumerate()
            .fold(0, |acc, (i, u)| {
                acc + (26usize.pow(i as u32)) * (u as usize - 64)
            }) as i32
    }
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test_1() {
        assert_eq!(Solution::title_to_number("A".to_string()), 1);
    }

    #[test]
    fn test_2() {
        assert_eq!(Solution::title_to_number("AB".to_string()), 28);
    }

    #[test]
    fn test_3() {
        assert_eq!(Solution::title_to_number("ZY".to_string()), 701);
    }

    #[test]
    fn test_4() {
        assert_eq!(Solution::title_to_number("FXSHRXW".to_string()), 2147483647);
    }
}

File: leetcode/pyleetcode/pyleetcode/easy/excel_sheet_column_number.py
"""171. Excel Sheet Column Number

https://leetcode.com/problems/excel-sheet-column-number/
"""

from typing import Optional, List, Any  # noqa: F401


class Solution:

    def titleToNumber(self, columnTitle: str) -> int:
        # res = 0
        # for i, ch in enumerate(reversed(columnTitle)):
        #     res += (26 ** i) * (ord(ch) - 64)
        # return res
        return sum(((26 ** i) * (ord(ch) - 64) for i, ch
                    in enumerate(reversed(columnTitle))), 0)

    def process(self, *args, **kwargs):  # noqa: N802
        return self.titleToNumber(*args, **kwargs)
